combat-trainer.lic

```diff
@@ -246,10 +249,29 @@ class LootProcess
     game_state.mob_died = true
     arrange_mob(DRRoom.dead_npcs.first, game_state)
     if @necro
-      if @ritual_type == 'harvest'
+      if @heal
+        wounds = check_wounds
+        echo "Wounds: #{wounds}" if $debug_mode_ct
+        unless wounds.empty?
+          Flags.add('necro-spellcast', '^You feel fully prepared to cast your spell\.')
+          prev_ritual = @ritual_type
+          @ritual_type = 'consume'
+          echo "Temporarily setting ritual to #{@ritual_type}" if $debug_mode_ct
+          bput('release', '')
+          bput("prepare cf", @settings.prep_messages)
+        end
+      end
+      if @ritual_type == ('harvest' || 'consume')
         check_thanatology(DRRoom.dead_npcs.first, 'preserve', game_state)
       end
       check_thanatology(DRRoom.dead_npcs.first, @ritual_type, game_state)
+      if @heal && @ritual_type == 'consume'
+        pause 0.5 until Flags['necro-spellcast']
+        fput("cast #{wounds.max_by{|k,v| v}[0]}")
```

Comment:
Ah, thank you. What about `wounds.key(wounds.values.max)`?

Although if you're sorting by the value to find a key, it makes me think that you should consider swapping your keys and values in the hash.